---
title: "Group 7 Final project"
subtitle: "Cell-type and brain region-resolved mouse proteome"
format:
  revealjs:
    fig-format: png
    execute:
      fig-device: ragg_png
editor: visual
---

```{r}
#| warning: false
library("tidyverse")
library("pcaMethods")
library("janitor")
library("patchwork")
library("ggrepel")

#Source functions
source("../R/99_proj_func.R")
```

# Introduction

Marianna

## 

![](images/clipboard-1107948132.png){width="3600"}

# Results

## Transcriptome density analysis

```{r}
knitr::include_graphics("../results/RNA_density_plot_celltype.png")

knitr::include_graphics("../results/RNA_density_plot_combined.png")
```

## Marianna

![](images/Screenshot%202025-12-01%20at%2019.58.33.png)

## Correlation Heatmap

```{r}
aug_3 <- read_tsv(file = "../data/augmented_data/table_3_log_means.tsv",
                  show_col_types = FALSE) |> 
  select(rna_oligo_div1:rna_neuron_div10, gene_name)

aug_5 <- read_tsv(file = "../data/augmented_data/table_5_log_means.tsv",
                  show_col_types = FALSE) |> 
  select(gene_name, protein_names, prot_microglia:prot_oligo_div4) 


prot_one_gene <- aug_5 |> 
  mutate(gene_name = str_split(gene_name, ";") |> map_chr(1))

prot_rna_merged <- inner_join(prot_one_gene, aug_3, by = "gene_name")

mat <- prot_rna_merged |>
  select(starts_with("prot_"), starts_with("rna_"))

prot_mat <- mat |> 
  select(starts_with("prot_")) |>
  as.matrix()

rna_mat  <- mat |>
  select(starts_with("rna_"))  |>
  as.matrix()

corr_mat <- cor(prot_mat, rna_mat, use = "pairwise.complete.obs")

corr_df <- corr_mat |> 
  as_tibble(rownames = "proteome") |> #then we produce the long format
  pivot_longer(-proteome,
               names_to = "transcriptome",
               values_to = "corr")


corr_df <- corr_df |>
  mutate( # we will use recode because it will preserve the existing order of levels while changing the values
    proteome = recode(proteome,
      prot_microglia    = "Microglia",
      prot_astro        = "Astrocytes",
      prot_oligo_div1   = "Oligo_DIV1",
      prot_oligo_div2_5 = "Oligo_DIV2.5",
      prot_oligo_div4   = "Oligo_DIV4",
      prot_neuron_div5  = "Neuron_DIV5",
      prot_neuron_div10 = "Neuron_DIV10"
    ),
    transcriptome = recode(transcriptome,
      rna_microglia    = "Microglia",
      rna_astro        = "Astrocytes",
      rna_oligo_div1   = "Oligo_DIV1",
      rna_oligo_div2_5 = "Oligo_DIV2.5",
      rna_oligo_div4   = "Oligo_DIV4",
      rna_neuron_div5  = "Neuron_DIV5",
      rna_neuron_div10 = "Neuron_DIV10"
    )
  )


heatmap_tr_prt <- ggplot(corr_df, aes(x = transcriptome, y = proteome, fill = corr)) +
  geom_tile() +
  geom_text(aes(label = sprintf("%.2f", corr)), size = 3) +
  scale_fill_gradient2(low = "steelblue", mid = "white", high = "darkred",
                       midpoint = 0.30) +
  theme_minimal() +
  theme(
    axis.text.x = element_text(angle = 45, hjust = 1),
    panel.grid = element_blank()
  ) +
  labs(
    title = "Correlation Between Transcriptome and Proteome",
    x = "Transcriptome (log2 RPKM)",
    y = "Proteome (log2 LFQ)",
    fill = "Pearson correlation"
  )

heatmap_tr_prt

```

## LogFold change across Cell types

```{r}
#Adding the markers of each cell type
astro_markers       <- c("Aqp4", "Gfap", "Glast", "Aldh1l1")
micro_markers       <- c("Iba1", "Tlr2", "Tlr7", "CD86")
neuron_markers      <- c("Snap25", "Tubb3", "Calb1", "Syt1")
oligo_markers       <- c("Cnp1", "Sox10", "Mbp", "Mag", "Mog", "Plp1")

all_df <- read_tsv("../data/augmented_data/table_2_log.tsv", show_col_types = FALSE)

#Selecting only the rows corresponding to marker genes.
#These points will be labeled and colored in the plot using geom_text_repel()

labels_df <- all_df |>
  group_by(type) |>
  filter(
    (type == "Astrocytes"       & gene %in% astro_markers)  |
    (type == "Microglia"        & gene %in% micro_markers)  |
    (type == "Neurons"          & gene %in% neuron_markers) |
    (type == "Oligodendrocytes" & gene %in% oligo_markers)
  ) |>
  ungroup()

#Customing color palette for each marker category

cols <- c(
  "Astrocyte markers"       = "#1A9850",
  "Microglia markers"       = "#2B83BA",
  "Neuron markers"          = "#00008B",
  "Oligodendrocyte markers" = "#D7191C",
  "Other"                   = "grey70"
)

# Base theme applied to all panels
base_theme <- theme_bw(base_size = 12) +
  theme(
    plot.title = element_text(face = "bold", size = 14),
    legend.title = element_blank(),
    legend.text  = element_text(size = 14)
  )



# Generating a panel for each cell type
# Each panel uses its subset of data and its marker labels

p_astro  <- draw_panel(all_df |> filter(type == "Astrocytes"),       labels_df |> filter(type == "Astrocytes"),       "Astrocytes")
p_micro  <- draw_panel(all_df |> filter(type == "Microglia"),        labels_df |> filter(type == "Microglia"),        "Microglia")
p_neuron <- draw_panel(all_df |> filter(type == "Neurons"),          labels_df |> filter(type == "Neurons"),          "Neurons")
p_oligo  <- draw_panel(all_df |> filter(type == "Oligodendrocytes"), labels_df |> filter(type == "Oligodendrocytes"), "Oligodendrocytes")

# Combining all four panels into a single figure
# Arranging them in 2 rows and place the legend at the bottom

combined <- p_astro + p_micro + p_neuron + p_oligo +
  plot_layout(nrow = 2, guides = "collect") &
  theme(legend.position = "bottom")

combined
```

## Principal Component Analysis

```{r}
#| warning: false

clean_regions <- read_tsv(file = "../data/clean_data/clean_17.tsv") 
clean_regions <- clean_regions |> 
    select(14:23) 

colnames(clean_regions) <- (c("Brainstem", "Cerebellum", "Corpus callosum", "Motor cortex", 
  "Olfactory bulb", "Optic nerve", "Prefrontal cortex", "Striatum", "Thalamus", 
  "Hippocampus"))

#############################################################################

#Create an object with ONLY numerical data for the PCA object
numerical_brain_regions <- clean_regions |> 
  select_if(map_lgl(.x = clean_regions, .f = is.numeric)) |> 
  remove_empty("rows") #Comepletely empty rows can't be used at all (this is said by checkData, used in the following steps)

#############################################################################

brdata <- numerical_brain_regions |> 
  prep(scale = "none", center = T) 

resPCA <- pca(brdata, method = "svd", center = F, nPcs = 5)

#############################################################################

#Get the PCA values
pcaData <- as.data.frame(loadings(resPCA)) |> 
  mutate(region = row.names(loadings(resPCA)))

#Get the % of variability captured by each PC
PC1var <- resPCA@R2[1] * 100
PC2var <- resPCA@R2[2] * 100

#Plot!
PCA_plot <- pcaData |> 
  ggplot(aes(x = PC1, y = PC2)) +
  geom_point(color = "blue", shape = 19) +
  geom_label_repel(aes(label = region),
                   size = 3) +
  labs(y = paste("Principal component 2 (", PC2var, "% variability )"),
       x = paste("Principal component 1 (", PC1var, "% variability )"),
       title = "PCA of proteome by brain region") +
  theme_minimal()

```

```{r}
PCA_plot
```

# Conclusion

-   **Transcriptomics â‰  Proteomics**

    -   Most abundant transcripts and the most abundant proteins often do not correlate

-   **Proteomics provides valuable insights**

    -   Different proteomic profiles depending on cell type / brain region

-   **Why is this important?**

    -   mRNA results don't translate into real world proteomic profiles
