---
title: "03_augment"
format: html
editor: visual
---

# Augmenting data

```{r}
library("tidyverse")

source("../R/99_proj_func.R")
```

## Table 2

```{r}
clean_2 <- read_tsv(file = "../data/clean_data/clean_2.tsv", show_col_types = FALSE)

#Computing medians and log fold change of the gene expression among different cell types

panel_df <- clean_2 |>
  rowwise() |>
  mutate(
    logLFQ_median = median(c_across(starts_with("logLFQ_")), na.rm = TRUE),
    logFC_astro   = logLFQ_astro  - logLFQ_median,
    logFC_micro   = logLFQ_micro  - logLFQ_median,
    logFC_neuron  = logLFQ_neuron - logLFQ_median,
    logFC_oligo   = logLFQ_oligo  - logLFQ_median
  ) |>
  ungroup()

#Constructing the table:

astro_df <- panel_df |> 
  select(gene, logLFQ = logLFQ_astro, logFC = logFC_astro) |>
  mutate(type = "Astrocytes")

micro_df <- panel_df |> 
  select(gene, logLFQ = logLFQ_micro, logFC = logFC_micro) |>
  mutate(type = "Microglia")

neuron_df <- panel_df |> 
  select(gene, logLFQ = logLFQ_neuron, logFC = logFC_neuron) |>
  mutate(type = "Neurons")

oligo_df <- panel_df |> 
  select(gene, logLFQ = logLFQ_oligo, logFC = logFC_oligo) |>
  mutate(type = "Oligodendrocytes")

all_df <- bind_rows(
  astro_df, micro_df, neuron_df, oligo_df
)

#Adding the markers of each cell type
astro_markers       <- c("Aqp4", "Gfap", "Glast", "Aldh1l1")
micro_markers       <- c("Iba1", "Tlr2", "Tlr7", "CD86")
neuron_markers      <- c("Snap25", "Tubb3", "Calb1", "Syt1")
oligo_markers       <- c("Cnp1", "Sox10", "Mbp", "Mag", "Mog", "Plp1")

all_df <- all_df |>
  mutate(
    category = case_when(
      gene %in% astro_markers  ~ "Astrocyte markers",
      gene %in% micro_markers  ~ "Microglia markers",
      gene %in% neuron_markers ~ "Neuron markers",
      gene %in% oligo_markers  ~ "Oligodendrocyte markers",
      TRUE                     ~ "Other"
    )
  ) |>
  filter(is.finite(logFC), is.finite(logLFQ))

# Writing the table to augmented data folder data/data_augmented in tsv format



```

## Table 3

```{r}
clean_3 <- read_tsv(file = "../data/clean_data/clean_3.tsv", show_col_types = FALSE)

```

```{r}
rna_prefixes <- c(
  rna_oligo_div1   = "rpkm_oligodendrocytes_div1_",
  rna_oligo_div2_5 = "rpkm_oligodendrocytes_div2_5_",
  rna_oligo_div4   = "rpkm_oligodendrocytes_div4_",
  rna_microglia    = "rpkm_adult_microglia_",
  rna_astro        = "rpkm_astrocytes_",
  rna_neuron_div5  = "rpkm_cortical_neurons_div05_",
  rna_neuron_div10 = "rpkm_cortical_neurons_div10_"
)
```

The values in the plots are **log2 transformed** so that's the transformation we are going to apply to proteome and transcriptome. We will use the **log2p1** function from 99_proj_func.R.

Also we can see that the there are cell types used im most of the plots. In the table there are 3 measurements for each cell type (adult_microglia_1, adult_microglia_2, adult_microglia_3). To get one value to compare cell types we will **take the mean**. That way we can get a more reliable estimate for each cell type by getting rid of experimental bias in each measurement and random noise. We are doing it using the **mean_across_col** function from 99_proj_func.R.

```{r}
aug_3 <- clean_3 |>
  mutate(across(starts_with("rpkm_"), log2p1)) |>  # log2(x+1) on all rpkm columns
  mean_across_col(rna_prefixes) #|>                 # calculate means per cell type
  # as an output we will have the old columns + mean columns for each cell type
```

## Table 5

```{r}
clean_5 <- read_tsv(file = "../data/clean_data/clean_5.tsv", show_col_types = FALSE)
```

```{r}
prot_prefixes <- c(
  prot_microglia    = "adult_microglia_",
  prot_astro        = "astrocytes_",
  prot_neuron_div5  = "neurons_div05_",
  prot_neuron_div10 = "neurons_div10_",
  prot_oligo_div1   = "oligodendrocytes_div1_",
  prot_oligo_div2_5 = "oligodendrocytes_div2_5_",
  prot_oligo_div4   = "oligodendrocytes_div4_"
)
```

We will do the same transformations to table 5 as we did to table 3.

```{r}
aug_5 <- clean_5 |> 
  mutate(across(adult_microglia_1:oligodendrocytes_div4_3, log2p1)) |> # applying log2
  mean_across_col(prot_prefixes) |> #df is provided by pipe
  mutate(gene_name = gene_names) #|> #renaming the column for future merging with rna df
  # as an output we will have the old columns + mean columns for each cell type

```

```{r}
dir.create("../data/augmented_data", recursive = TRUE, showWarnings = FALSE)

write_tsv(all_df, "../data/augmented_data/table_2_log.tsv")

write_tsv(aug_3, "../data/augmented_data/table_3_log_means.tsv")

write_tsv(aug_5, "../data/augmented_data/table_5_log_means.tsv")
```
