---
title: "2c_plot"
format: html
---

# The **heatmap between proteome and transcriptome** across cell types.

```{r}
library("tidyverse")
library("ggplot2")
```

## Loading the data

```{r}
```

```{r}
```

```{r}
```

```{r}
```

```{r}
aug_3 <- read_tsv(file = "../data/augmented_data/table_3_log_means.tsv", show_col_types = FALSE) |> 
  select(rna_oligo_div1:rna_neuron_div10, gene_name)

aug_5 <- read_tsv(file = "../data/augmented_data/table_5_log_means.tsv", show_col_types = FALSE) |> 
  select(gene_name, protein_names, prot_microglia:prot_oligo_div4) 

```

## Transforming the data for plotting

### Proteome data

The values in the plot are log2 transformed so that's the transformation we are going to apply to proteome and transcriptome.

Also we can see that the on the X and Y axis there are cell types. In the table there are 3 measurements for each cell type (adult_microglia_1, adult_microglia_2, adult_microglia_3). To get one value to compare cell types we will take the mean. That way we can get a more reliable estimate for each cell type by getting rid of experimental bias in each measurement and random noise.

```{r}
#log2p1 <- function(x) {log2(x + 1)}
```

```{r}

# prot_for_plot <- clean_5 |>
#   mutate(across(adult_microglia_1:oligodendrocytes_div4_3, log2p1)) |> # applying log2
#   rowwise() |> # dplyr function that applies the next function to each row, otherwise it would be applied to columns
#   mutate( 
#     prot_microglia   = mean(c_across(starts_with("adult_microglia_")),        na.rm = TRUE), # ignoring missing values
#     prot_astro       = mean(c_across(starts_with("astrocytes_")),             na.rm = TRUE), # c_across is c(across())
#     prot_neuron_div5 = mean(c_across(starts_with("neurons_div05_")),          na.rm = TRUE),
#     prot_neuron_div10= mean(c_across(starts_with("neurons_div10_")),          na.rm = TRUE),
#     prot_oligo_div1  = mean(c_across(starts_with("oligodendrocytes_div1_")),  na.rm = TRUE),
#     prot_oligo_div2_5= mean(c_across(starts_with("oligodendrocytes_div2_5_")),na.rm = TRUE),
#     prot_oligo_div4  = mean(c_across(starts_with("oligodendrocytes_div4_")),  na.rm = TRUE),
#     gene_name = gene_names
#   ) |>
#   ungroup() |> # here we remove row-wise grouping
  # select(gene_name, protein_names, prot_microglia:prot_oligo_div4)
```

```{r}
# prot_prefixes <- c(
#   prot_microglia    = "adult_microglia_",
#   prot_astro        = "astrocytes_",
#   prot_neuron_div5  = "neurons_div05_",
#   prot_neuron_div10 = "neurons_div10_",
#   prot_oligo_div1   = "oligodendrocytes_div1_",
#   prot_oligo_div2_5 = "oligodendrocytes_div2_5_",
#   prot_oligo_div4   = "oligodendrocytes_div4_"
# )
```

```{r}
# prot_for_plot <- clean_5 |> 
#   mutate(across(adult_microglia_1:oligodendrocytes_div4_3, log2p1)) |> # applying log2
#   mean_across_col(prot_prefixes) |> #df is provided by pipe
#   mutate(gene_name = gene_names) |> #renaming the column for future merging with rna df
#   select(gene_name, protein_names, prot_microglia:prot_oligo_div4) #saving only new cols and names 
#   
# prot_for_plot
```

```{r}
# writing a table to augmented data folder data/data_augmented in tsv format
# dir.create("../data/augmented_data", recursive = TRUE, showWarnings = FALSE)
# 
# write_tsv(prot_for_plot, "../data/augmented_data/table_5_log_means.tsv")
```

### Transcriptome data

```{r}
# rna_for_plot <- clean_3 |>
#   mutate(across(starts_with("rpkm_"), log2p1)) |>   
#   mean_across_col(prot_prefixes) |> 
#   select(rna_oligo_div1:rna_neuron_div10, gene_name)
# 
# rna_for_plot 
```

```{r}
```

## Merging the tables

First of all we need to merge two datasets on gene_name

In the initial table we had multiple gene names in one cell. What does this mean? Mass spectrometry measures short peptides and proteins can have the same peptide sequences. So we cannot say for sure which protein and which gene the peptide came from that's why we list multiple genes. Also in the gene name column there can be different name notations.

If we put the gene names from one protein to different rows, we create fake disagreement between RNA and proteins beacuse even if the protein is highly presented the RNA for some of the genes in its group can be very low.\
Also when we replicate one gene multiple times, we create correlated rows, which violates the mathematical assumptions of Pearson correlation.

```{r}
#prot_rna_merged_00 <- inner_join(prot_for_plot, rna_for_plot, by = "gene_name")
#prot_rna_merged_00
```

```{r}
#any(duplicated(prot_rna_merged_00$protein_names)) # check duplictes in protein names column 
```

### Keeping only one gene per protein

So the reasonable thing to do here is to keep one gene in each protein raw, we will keep the first gene name extracted from the sequence of genes.

A protein group may be identified because the detected peptides match more than one protein-coding gene. These genes are listed together in the gene_name column.

Why the first gene in the annotation is the correct one to keep? In proteomics tools the first gene listed represents the “leading protein” of the protein group. Even if the table shows one row for one protein, that protein may be the representative of a protein group.

```{r}
```

```{r}
prot_one_gene <- aug_5 |> 
  mutate(gene_name = str_split(gene_name, ";") |> map_chr(1)) #extracting the first gene and keeping it
#prot_one_gene
```

### Merging

```{r}
```

```{r}
# any(duplicated(clean_5$protein_names))
```

```{r}
# clean_5 |> 
#   filter(duplicated(protein_names) | duplicated(protein_names, fromLast = TRUE))

#So there are duplicates in protein names initially and its okay
```

```{r}
prot_rna_merged <- inner_join(prot_one_gene, aug_3, by = "gene_name")
prot_rna_merged # joined table with duplicates in protein names and it okay, no corr in data
```

## Calculation of correlation

```{r}

mat <- prot_rna_merged |>
  select(starts_with("prot_"), starts_with("rna_"))

prot_mat <- mat |> select(starts_with("prot_")) |> as.matrix()
rna_mat  <- mat |> select(starts_with("rna_"))  |> as.matrix()

corr_mat <- cor(prot_mat, rna_mat, use = "pairwise.complete.obs")
corr_mat
```

```{r}
# but for plotting we need a table not a matrix
corr_df <- corr_mat |> 
  as_tibble(rownames = "proteome") |> #then we produce the long format
  pivot_longer(-proteome,
               names_to = "transcriptome",
               values_to = "corr")
corr_df
```

```{r}

```

```{r}
corr_df <- corr_df |>
  mutate( # we will use recode because it will preserve the existing order of levels while changing the values.
    proteome = recode(proteome,
      prot_microglia    = "Microglia",
      prot_astro        = "Astrocytes",
      prot_oligo_div1   = "Oligo_DIV1",
      prot_oligo_div2_5 = "Oligo_DIV2.5",
      prot_oligo_div4   = "Oligo_DIV4",
      prot_neuron_div5  = "Neuron_DIV5",
      prot_neuron_div10 = "Neuron_DIV10"
    ),
    transcriptome = recode(transcriptome,
      rna_microglia    = "Microglia",
      rna_astro        = "Astrocytes",
      rna_oligo_div1   = "Oligo_DIV1",
      rna_oligo_div2_5 = "Oligo_DIV2.5",
      rna_oligo_div4   = "Oligo_DIV4",
      rna_neuron_div5  = "Neuron_DIV5",
      rna_neuron_div10 = "Neuron_DIV10"
    )
  )
```

```{r}
heatmap_tr_prt <- ggplot(corr_df, aes(x = transcriptome, y = proteome, fill = corr)) +
  geom_tile() +
  geom_text(aes(label = sprintf("%.2f", corr)), size = 3) +
  scale_fill_gradient2(low = "steelblue", mid = "white", high = "darkred",
                       midpoint = 0.30) +
  theme_minimal() +
  theme(
    axis.text.x = element_text(angle = 45, hjust = 1),
    panel.grid = element_blank()
  ) +
  labs(
    title = "Correlation Between Transcriptome and Proteome",
    x = "Transcriptome (log2 RPKM)",
    y = "Proteome (log2 LFQ)",
    fill = "Pearson correlation"
  )

heatmap_tr_prt

```

```{r}
ggsave(filename = "Heatmap_tr_prt.jpg", plot = heatmap_tr_prt, path = "../results", width = 8, height = 6)
```
