---
title: "2c_plot"
format: html
---

# The **heatmap between proteome and transcriptome** across cell types.

```{r}
library("tidyverse")
library("ggplot2")
```

## Loading the data

```{r}
aug_3 <- read_tsv(file = "../data/augmented_data/table_3_log_means.tsv",
                  show_col_types = FALSE) |> 
  select(rna_oligo_div1:rna_neuron_div10, gene_name)

aug_5 <- read_tsv(file = "../data/augmented_data/table_5_log_means.tsv",
                  show_col_types = FALSE) |> 
  select(gene_name, protein_names, prot_microglia:prot_oligo_div4) 

```

## Merging the tables

First of all we need to merge two datasets on gene_name

In the initial table we had multiple gene names in one cell. Mass spectrometry measures short peptides and proteins can have the same peptide sequences. So we cannot say for sure which protein and which gene the peptide came from that's why we list multiple genes. Also in the gene name column there can be different name notations.

If we put the gene names from one protein to different rows, we create fake disagreement between RNA and proteins beacuse even if the protein is highly presented the RNA for some of the genes in its group can be very low.\
Also when we replicate one gene multiple times, we create correlated rows, which violates the mathematical assumptions of Pearson correlation.

### Keeping only one gene per protein

So the reasonable thing to do here is to keep one gene in each protein raw, we will keep the first gene name extracted from the sequence of genes.

Why the first gene in the annotation is the correct one to keep? In proteomics tools the first gene listed represents the “leading protein” of the protein group. Even if the table shows one row for one protein, that protein may be the representative of a protein group.

```{r}
prot_one_gene <- aug_5 |> 
  mutate(gene_name = str_split(gene_name, ";") |> map_chr(1)) #extracting the first gene and keeping it
#prot_one_gene
```

### Merging

```{r}
prot_rna_merged <- inner_join(prot_one_gene, aug_3, by = "gene_name")
prot_rna_merged # joined table with duplicates in protein names and it's okay, no correlation in data
```

## Calculation of correlation

```{r}
mat <- prot_rna_merged |>
  select(starts_with("prot_"), starts_with("rna_"))

prot_mat <- mat |> 
  select(starts_with("prot_")) |>
  as.matrix()

rna_mat  <- mat |>
  select(starts_with("rna_"))  |>
  as.matrix()

corr_mat <- cor(prot_mat, rna_mat, use = "pairwise.complete.obs")
corr_mat
```

```{r}
# but for plotting we need a table not a matrix
corr_df <- corr_mat |> 
  as_tibble(rownames = "proteome") |> #then we produce the long format
  pivot_longer(-proteome,
               names_to = "transcriptome",
               values_to = "corr")
corr_df
```

```{r}
corr_df <- corr_df |>
  mutate( # we will use recode because it will preserve the existing order of levels while changing the values
    proteome = recode(proteome,
      prot_microglia    = "Microglia",
      prot_astro        = "Astrocytes",
      prot_oligo_div1   = "Oligo_DIV1",
      prot_oligo_div2_5 = "Oligo_DIV2.5",
      prot_oligo_div4   = "Oligo_DIV4",
      prot_neuron_div5  = "Neuron_DIV5",
      prot_neuron_div10 = "Neuron_DIV10"
    ),
    transcriptome = recode(transcriptome,
      rna_microglia    = "Microglia",
      rna_astro        = "Astrocytes",
      rna_oligo_div1   = "Oligo_DIV1",
      rna_oligo_div2_5 = "Oligo_DIV2.5",
      rna_oligo_div4   = "Oligo_DIV4",
      rna_neuron_div5  = "Neuron_DIV5",
      rna_neuron_div10 = "Neuron_DIV10"
    )
  )
```

```{r}
heatmap_tr_prt <- ggplot(corr_df, aes(x = transcriptome, y = proteome, fill = corr)) +
  geom_tile() +
  geom_text(aes(label = sprintf("%.2f", corr)), size = 3) +
  scale_fill_gradient2(low = "steelblue", mid = "white", high = "darkred",
                       midpoint = 0.30) +
  theme_minimal() +
  theme(
    axis.text.x = element_text(angle = 45, hjust = 1),
    panel.grid = element_blank()
  ) +
  labs(
    title = "Correlation Between Transcriptome and Proteome",
    x = "Transcriptome (log2 RPKM)",
    y = "Proteome (log2 LFQ)",
    fill = "Pearson correlation"
  )

heatmap_tr_prt

```

```{r}
# save the plot to results folder
ggsave(filename = "Heatmap_tr_prt.jpg", plot = heatmap_tr_prt, path = "../results", width = 8, height = 6)
```
