---
title: "2c_plot"
format: html
---

# Plot 2C. The **heatmap between proteome and transcriptome** across cell types.

## Loading the data

```{r}
dirty_5 <- read_tsv(
  "../data/dirty_data/supp_5.tsv",
  skip = 1,          # skipping the first row (states LFQ)
  col_names = TRUE   # use the second row as header
)
```

```{r}
view(dirty_5)
```

```{r}
clean_5 <- dirty_5 |> 
  select(-contains("young"), -PEP, - `Mol. weight [kDa]`, -`Protein IDs`, -`Majority protein IDs`) |> 
  clean_names() |> 
  drop_na() |> 
  distinct(gene_names, .keep_all = T) 

#This table is with the separated gene names on each column. 
#"Tidy data" but would mess up our numeric operations. 
clean_5_separate_gene_names <- clean_5 |> 
  separate_longer_delim(cols = gene_names, delim = ";")
View(clean_5_separate_gene_names)
```

```{r}
clean_3 <- read_tsv(file = "../data/clean_data/clean_3.tsv", show_col_types = FALSE)
View(clean_3)
```

```{r}
```

## Transforming the data for plotting

### Proteome data

The values in the plot are log2 transformed so that's the transformation we are going to apply to proteome and transcriptome

Also we can see that the on the X and Y axis there are cell types. In the table there are 3 measurements for each cell type (adult_microglia_1, adult_microglia_2, adult_microglia_3). To get one value to compare cell types we will take the mean. That way we can get a more reliable estimate for each cell type by getting rid of experimental bias in each measurement and random noise.

```{r}
log2p1 <- function(x) {log2(x + 1)}
```

```{r}
#log2p1 <- function(x) {log2(x + 1)}

prot_for_plot <- clean_5 |>
  mutate(across(adult_microglia_1:oligodendrocytes_div4_3, log2p1)) |> # applying log2
  rowwise() |> # dplyr function that applies the next function to each row, otherwise it would be applied to columns
  mutate( 
    prot_microglia   = mean(c_across(starts_with("adult_microglia_")),        na.rm = TRUE), # ignoring missing values
    prot_astro       = mean(c_across(starts_with("astrocytes_")),             na.rm = TRUE), # c_across is c(across())
    prot_neuron_div5 = mean(c_across(starts_with("neurons_div05_")),          na.rm = TRUE),
    prot_neuron_div10= mean(c_across(starts_with("neurons_div10_")),          na.rm = TRUE),
    prot_oligo_div1  = mean(c_across(starts_with("oligodendrocytes_div1_")),  na.rm = TRUE),
    prot_oligo_div2_5= mean(c_across(starts_with("oligodendrocytes_div2_5_")),na.rm = TRUE),
    prot_oligo_div4  = mean(c_across(starts_with("oligodendrocytes_div4_")),  na.rm = TRUE),
    gene_name = gene_names
  ) |>
  ungroup() |> # here we remove row-wise grouping
  select(gene_name, protein_names, prot_microglia:prot_oligo_div4)
```

```{r}
prot_for_plot
```

### Transcriptome data

```{r}
rna_for_plot <- clean_3 |>
  mutate(across(starts_with("rpkm_"), log2p1)) |>   
  rowwise() |>
  mutate(
    rna_oligo_div1   = mean(c_across(starts_with("rpkm_oligodendrocytes_div1_")),   na.rm = TRUE),
    rna_oligo_div2_5 = mean(c_across(starts_with("rpkm_oligodendrocytes_div2_5_")), na.rm = TRUE),
    rna_oligo_div4   = mean(c_across(starts_with("rpkm_oligodendrocytes_div4_")),   na.rm = TRUE),
    rna_microglia    = mean(c_across(starts_with("rpkm_adult_microglia_")),         na.rm = TRUE),
    rna_astro        = mean(c_across(starts_with("rpkm_astrocytes_")),              na.rm = TRUE),
    rna_neuron_div5  = mean(c_across(starts_with("rpkm_cortical_neurons_div05_")),  na.rm = TRUE),
    rna_neuron_div10 = mean(c_across(starts_with("rpkm_cortical_neurons_div10_")),  na.rm = TRUE)
  ) |>
  ungroup() |> 
  select(rna_oligo_div1:rna_neuron_div10, gene_name)
```

```{r}
rna_for_plot 
```

## Merging the tables

First of all we need to merge two datasets on gene_name

In the initial table we had multiple gene names in one cell. What does this mean? Mass spectrometry measures short peptides and proteins can have the same peptide sequences. So we cannot say for sure which protein and which gene the peptide came from that's why we list multiple genes. Also in the gene name column there can be different name notations.

But this bad for correlation...We create fake disagreement between RNA and proteins beacuse even if the protein is highly presented the RNA for some of the genes in its group can be very low.\
Also when we replicate one gene multiple times, we create correlated rows, which violates the mathematical assumptions of Pearson correlation.

#First lets try to merge and see if we save multiple genes from a gene group after inner joining with rna table.

```{r}
#prot_rna_merged_00 <- inner_join(prot_for_plot, rna_for_plot, by = "gene_name")
#prot_rna_merged_00
```

```{r}
#any(duplicated(prot_rna_merged_00$protein_names)) # check duplictes in protein names column 
```

#So multiple genes from one gene group and with one protein name were present in RNA table. That is a bad thing because some of the #rows in our merged table are strongly correlated. So we should keep only one gene per protein.

### Keeping only one gene per protein

```{r}
prot_for_plot <- clean_5 |>
  mutate(across(adult_microglia_1:oligodendrocytes_div4_3, log2p1)) |> # applying log2
  rowwise() |> # dplyr function that applies the next function to each row, otherwise it would be applied to columns
  mutate( 
    prot_microglia   = mean(c_across(starts_with("adult_microglia_")),        na.rm = TRUE), # ignoring missing values
    prot_astro       = mean(c_across(starts_with("astrocytes_")),             na.rm = TRUE), # c_across is c(across())
    prot_neuron_div5 = mean(c_across(starts_with("neurons_div05_")),          na.rm = TRUE),
    prot_neuron_div10= mean(c_across(starts_with("neurons_div10_")),          na.rm = TRUE),
    prot_oligo_div1  = mean(c_across(starts_with("oligodendrocytes_div1_")),  na.rm = TRUE),
    prot_oligo_div2_5= mean(c_across(starts_with("oligodendrocytes_div2_5_")),na.rm = TRUE),
    prot_oligo_div4  = mean(c_across(starts_with("oligodendrocytes_div4_")),  na.rm = TRUE),
    gene_name = gene_names
  ) |>
  ungroup() |> # here we remove row-wise grouping
  select(gene_name, protein_names, prot_microglia:prot_oligo_div4)
```

```{r}
prot_one_gene <- prot_for_plot |> 
  mutate(gene_name = str_split(gene_name, ";") |> map_chr(1)) #extracting the first gene and keeping it
prot_one_gene
```

### Merging without duplicates

```{r}

```

```{r}
any(duplicated(clean_5$protein_names))
```

```{r}
clean_5 |> 
  filter(duplicated(protein_names) | duplicated(protein_names, fromLast = TRUE))
```

So there are duplicates in protein names initially and its okay

```{r}
prot_rna_merged_01 <- inner_join(prot_one_gene, rna_for_plot, by = "gene_name")
prot_rna_merged_01 # joined table with duplicates in protein names and it okay, no corr in data
```

## Calculation correlation

```{r}

mat <- prot_rna_merged_01 |>
  select(starts_with("prot_"), starts_with("rna_"))

prot_mat <- mat |> select(starts_with("prot_")) |> as.matrix()
rna_mat  <- mat |> select(starts_with("rna_"))  |> as.matrix()

corr_mat <- cor(prot_mat, rna_mat, use = "pairwise.complete.obs")

# but for plotting we need a table not a matrix
corr_df <- corr_mat |> 
  as_tibble(rownames = "proteome") |> #then we produce the long format
  pivot_longer(-proteome,
               names_to = "transcriptome",
               values_to = "corr")
```

```{r}
corr_df
```

```{r}
corr_df <- corr_df |>
  mutate( # we will use recode because it will preserve the existing order of levels while changing the values.
    proteome = recode(proteome,
      prot_microglia    = "Microglia",
      prot_astro        = "Astrocytes",
      prot_oligo_div1   = "Oligo_DIV1",
      prot_oligo_div2_5 = "Oligo_DIV2.5",
      prot_oligo_div4   = "Oligo_DIV4",
      prot_neuron_div5  = "Neuron_DIV5",
      prot_neuron_div10 = "Neuron_DIV10"
    ),
    transcriptome = recode(transcriptome,
      rna_microglia    = "Microglia",
      rna_astro        = "Astrocytes",
      rna_oligo_div1   = "Oligo_DIV1",
      rna_oligo_div2_5 = "Oligo_DIV2.5",
      rna_oligo_div4   = "Oligo_DIV4",
      rna_neuron_div5  = "Neuron_DIV5",
      rna_neuron_div10 = "Neuron_DIV10"
    )
  )
```

```{r}
ggplot(corr_df, aes(x = transcriptome, y = proteome, fill = corr)) +
  geom_tile() + #draws one rectangle per row
  geom_text(aes(label = sprintf("%.2f", corr)), size = 3) +
  scale_fill_gradient2(low = "steelblue", mid = "white", high = "darkred",
                       midpoint = 0.35) +
  theme_minimal() +
  theme(
    axis.text.x = element_text(angle = 45, hjust = 1),
    panel.grid = element_blank()
  ) +
  labs(
    x = "Transcriptome (log2 RPKM)",
    y = "Proteome (log2 LFQ)",
    fill = "Pearson r"
  )

```
